src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -399,61 +417,29 @@ private static boolean isSingleLineIf(DetailAST literalIf) {
      */
     private static boolean isSingleLineLambda(DetailAST lambda) {
         final DetailAST block = lambda.getLastChild();
-        return lambda.getLineNo() == block.getLineNo();
+        return isOnSameLine(lambda, block);
     }
 
     /**
-     * Checks if current case statement is single-line statement, e.g.:
+     * Checks if switch member (case or default statement) is single-line statement, e.g.:
      * <p>
      * {@code
      * case 1: doSomeStuff(); break;
      * case 2: doSomeStuff(); break;
      * case 3: ;
+     * default: doSomeStuff();break;
      * }
      * </p>
-     * @param literalCase {@link TokenTypes#LITERAL_CASE case statement}.
-     * @return true if current case statement is single-line statement.
+     * @param ast {@link TokenTypes#LITERAL_CASE case statement} or
+     * {@link TokenTypes#LITERAL_DEFAULT default statement}.
+     * @return true if current switch member is single-line statement.
      */
-    private static boolean isSingleLineCase(DetailAST literalCase) {
-        boolean result = false;
-        final DetailAST slist = literalCase.getNextSibling();
-        if (slist == null) {
-            result = true;
-        }
-        else {
-            final DetailAST caseBreak = slist.findFirstToken(TokenTypes.LITERAL_BREAK);
-            if (caseBreak != null) {
-                final DetailAST block = slist.getFirstChild();
-                final boolean atOneLine = literalCase.getLineNo() == block.getLineNo();
-                result = atOneLine && block.getLineNo() == caseBreak.getLineNo();
-            }
-        }
-        return result;
-    }
-
-    /**
-     * Checks if current default statement is single-line statement, e.g.:
-     * <p>
-     * {@code
-     * default: doSomeStuff();
-     * }
-     * </p>
-     * @param literalDefault {@link TokenTypes#LITERAL_DEFAULT default statement}.
-     * @return true if current default statement is single-line statement.
-     */
-    private static boolean isSingleLineDefault(DetailAST literalDefault) {
-        boolean result = false;
-        final DetailAST slist = literalDefault.getNextSibling();
-        if (slist == null) {
-            result = true;
-        }
-        else {
-            final DetailAST block = slist.getFirstChild();
-            if (block != null && block.getType() != TokenTypes.SLIST) {
-                result = literalDefault.getLineNo() == block.getLineNo();
-            }
-        }
-        return result;
+    private static boolean isSingleLineSwitchMember(DetailAST ast) {
+        return Optional.of(ast)
+                .map(DetailAST::getNextSibling)
```

Comment:
if this was just refactoring, please try to move this refactoring to separate commit (first commit) of the same PR.
To make it clear in commit that you just do refactoring that does not change a logic.

as second commit please try to make changes that are required to fix logic, so by change it will be clear what was wrong and how it become correct.

Note: functional approach is ok, but keep in mind that with such approach code coverage will not help us to put corner cases to input files. Do not overuse functional style, in functinal style code is shorter, but if project 100% coverage and 100% pitest, quality validation will decrease, as nothing help to point on cases that are not covered.
Previously resolving uncovered code usually result in extension of Inputs, so code base of inputs grown, so quality was ok.

Reply:
It is not kind of refactoring that can be done separately, this things are related with issue.
The only separate refactoring is extracting `isOnSameLine` method, but I doubt it requires additional commit.

About functional style - I understand it. But in some cases it can be better, e.g. when we need to go 2-3 or  more levels down the ast with nullable children.

Reply:
np, just keep it in mind that functional style is not good by all means. There are still no tool in our CI that can verify that removal of one item from chain-call results in failure of test or some calls of map is "skipped".